client/src/main/java/io/getstream/chat/android/client/helpers/AttachmentHelper.kt

```diff
@@ -0,0 +1,31 @@
+package io.getstream.chat.android.client.helpers
+
+import io.getstream.chat.android.client.models.Attachment
+import io.getstream.chat.android.client.utils.SystemTimeProvider
+import org.apache.commons.validator.routines.UrlValidator
+
+class AttachmentHelper(private val systemTimeProvider: SystemTimeProvider = SystemTimeProvider()) {
+
+    private val urlValidator = UrlValidator()
+
+    fun hasValidUrl(attachment: Attachment): Boolean {
+        val url = attachment.url
+        return when {
+            url == null -> false
+            urlValidator.isValid(url).not() -> false
+            url.contains(KEY_WORD_EXPIRES) -> {
+                parseTimeStampOrNull(url)?.let { timestamp -> timestamp > systemTimeProvider.provideTime() } ?: false
+            }
+            else -> true
+        }
+    }
+
+    private fun parseTimeStampOrNull(url: String): Long? {
+        return url.substringAfter(KEY_WORD_EXPIRES).substringBefore(CHAR_AMPERSAND).toLongOrNull()
```

Comment:
We shouldn't do string manipulation for URL parsing, we can do this through utilities, it's kind of dangerous (makes too many assumptions about the structure of the URL).

Creating an [`android.net.Uri`](https://developer.android.com/reference/android/net/Uri) with `Uri.parse(url)` and then using methods like [`getQueryParameter`](https://developer.android.com/reference/android/net/Uri#getQueryParameter(java.lang.String)) can help us check if we have the query param is present, and what its value is.

Reply:
Thanks for point! Replaced this logic by logic of okhttp library.